### ISSUE TITLE
Scalar Input Output usage Angular 

### ISSUE THREAD

**Issue by burner** [dotansimha/graphql-code-generator#9492]
### Describe the bug

https://github.com/dotansimha/graphql-code-generator/pull/9375  broke my code in angular 16.

I have a query 
```
query GetDataAt($personId: Int!, $time: DateTime!) {
```

And I was able to call it like this.
```		
const vars = { personId: theId, time: theTime };
		this.theData = new GetDataAtGQL(this.apollo)
			.watch(vars)
```

But now typescript is telling me 

```
Argument of type '{ personId: { input: number; }; time: Date; }' is not assignable to parameter of type 'Exact<{ personId: { input: number; output: number; }; time: { input: any; output: any; }; }>'.
  Types of property 'personId' are incompatible.
    Property 'output' is missing in type '{ input: number; }' but required in type '{ input: number; output: number; }'.
```

Which technically is correct, but is quite annoying as I'm using the variables as input and not as output.

Am I doing something wrong here?

### Your Example Website or App

https://github.com/dotansimha/graphql-code-generator/pull/9375

### Steps to Reproduce the Bug or Issue

any graphql query with arguments on angular

### Expected behavior

to not have to do the conversion by hand.

Maybe by having the scalar definition maybe read

```ts
type intScalar = number | { input: number, output: number | string };
```

### Platform

- OS: [e.g. macOS, Windows, Linux]
- NodeJS: [e.g. 18.5.0]
- `graphql` version: [e.g. 16.3.0]
- `@graphql-codegen/*` version(s): [e.g. 2.6.2]

**Comment by eddeee888** [comment]
Hi @burner, do you mind posting your Codegen Config please and/or a sandbox example please? 🙂

**Comment by synoptase** [comment]
Same issue. Here's my config:
```ts
const generateConfig = async (): Promise<CodegenConfig> => {
  return {
    overwrite: true,
    schema: [
      {
        [process.env.HASURA_GRAPHQL_URL!]: {
          headers: {
            "x-hasura-admin-secret": process.env.HASURA_SECRET!,
          },
        },
      },
    ],
    documents: ["./src/store/api/schema.graphql"],
    generates: {
      "./src/store/api/generated.ts": {
        plugins: [
          "typescript",
          "typescript-operations",
          {
            "typescript-rtk-query": {
              importBaseApiFrom: "./base-api",
              exportHooks: true,
            },
          },
        ],
      },
    },
  };
};
```

**Comment by Drivez** [comment]
Hi I do have the same issue using React when updating to the latest version.
how can i revert and just use the standard types and not these `{ input: string, output: string }`

Config:


```
....
generates:
    src/generated/schema.json:
        plugins:
            - introspection
        config:
            minify: true
    src/generated/possibleTypes.ts:
        plugins:
            - fragment-matcher
        config:
            scalars:
                JsonObject: Record<string, unknown>
                Timestamp: Date
    src/generated/types.ts:
        plugins:
            - typescript
        config:
            scalars:
                JsonObject: Record<string, unknown>
                Timestamp: Date
    src/:
        preset: near-operation-file-preset
        presetConfig:
            baseTypesPath: generated/types.ts
            extension: .ts
            folder: generated
        plugins:
            - typescript-operations
            - typescript-react-apollo
        config:
            scalars:
                JsonObject: Types.Scalars['JsonObject']
                Timestamp: Types.Scalars['Timestamp']
            withHooks: false
```

**Comment by adevine** [comment]
@Drivez , see this comment and subsequent thread on the PR for this change, https://github.com/dotansimha/graphql-code-generator/pull/9375#issuecomment-1580389964.

In my opinion, especially since https://github.com/dotansimha/graphql-code-generator/pull/9497 was released so now there are no default scalars that have separate input/output values, this was overall an ill-considered change and should be reverted.

**Comment by eddeee888** [comment]
Hello all 👋 

I have left a response about reverting input/output [here](https://github.com/dotansimha/graphql-code-generator/pull/9375#issuecomment-1603413081).

TLDR; Input/output format will not be reverted. 

This is a necessary change to allow the library to correctly express GraphQL Scalar behaviours. This is useful for cases such as creating and typing custom scalars.

--- 

Here's how you can migrate to the new input/output approach:

If you are directly importing and using `Scalars` in your app's non-generated code, you can find all references of `Scalars['YourScalar']` and add either `input` or `output` depending on the scalar flow/direction i.e. `Scalars['YourScalar']['input']` or `Scalars['YourScalar']['output']`. 

If the types of input and output are the same, you can pick either. However, I recommend using the right flow to future-proof in case your custom scalar implementation changes.

---

Now, if you are seeing the generated code referencing `Scalars` without `input` or `output`, I'm happy to take a look at your setup if you have a Sandbox (here's the [CodeSandbox template](https://codesandbox.io/s/github/dotansimha/graphql-code-generator-issue-sandbox-template?file=/package.json)). 🙂

**Comment by eddeee888** [comment]
Hi @chronicIntrovert , please help me understand the whole setup using this [Codesandbox](https://codesandbox.io/p/sandbox/github/dotansimha/graphql-code-generator-issue-sandbox-template/tree/master) template 🙏

**Comment by amkisko** [comment]
Got similar issue after upgrading @graphql-codegen/cli to ^5.0.2, seems that only downgrading is the option, even not possible to debug and and figure out how to fix the code.

@eddeee888 is it possible to have an examples how to use latest version with @apollo/client, especially the part with `useQuery` and `variables`. I don't understand how to use these `input` and `output` while I should provide single value for the request.

**Comment by eddeee888** [comment]
Hello 👋 

Here's an example setup with React (Vite) + Apollo Client + GraphQL Codegen: https://github.com/eddeee888/react-graphql-template:

1. Scalar should look like this:

```ts
// src/graphql-types.generated.ts
export type Scalars = {
  ID: { input: string; output: string };
  String: { input: string; output: string };
  Boolean: { input: boolean; output: boolean };
  Int: { input: number; output: number };
  Float: { input: number; output: number };
  jsonb: { input: any; output: any };
};
```

2. Generated operations should have `input` automatically [wired up](https://github.com/eddeee888/react-graphql-template/blob/8c6da0933c770647078b272b1ccf595dc2a221e4/src/App.generated.ts#L6-L8):

```ts
// src/App.generated.ts
export type PokemonsQueryVariables = Types.Exact<{
  limit: Types.Scalars["Int"]["input"];
}>;
```

3. We can just use [the hook directly](https://github.com/eddeee888/react-graphql-template/blob/8c6da0933c770647078b272b1ccf595dc2a221e4/src/App.tsx#L7-L11):

```ts
// src/App.tsx
const { loading, error, data } = usePokemonsQuery({
  variables: {
    limit: 10,
  },
});
```

--- 

I haven't used Angular much before, so please post an Angular reproducible repo here. I think I can help figure it out.  🙏

**Comment by amkisko** [comment]
Thanks for response and examples! Yesterday I got it magically working, losing the traces and understanding what is actually happening, it just started to work at some point after an hour of try outs to fix it, been mostly trying to pin point which part is breaking the compilation. In my case it's pure react with typescript.

**Comment by eddeee888** [comment]
Closing this as there hasn't been activities. Please let me know if this is still an issue.
I'd be happy to take a look at reproducible repos
